### DISCUSSION TITLE
Q.dialog plugin - Can a custom component be positioned using Q.dialog position option?

### DISCUSSION THREAD

**Discussion by kunle-e** [quasarframework/quasar#11395] [General - Components / Directives / etc] [answered]
Does anyone know if it is expected  to be able position  a custom component  when invoking it using a q.dialog by using the q.dialog's 'position' option? 
Options like position and transitions that are set on the q.dialog do not affect the custom component.

[https://quasar.dev/quasar-plugins/dialog#invoking-custom-component ](https://quasar.dev/quasar-plugins/dialog#invoking-custom-component )

**Comment by metalsadman** [comment, marked as answer]
yes you can. https://codepen.io/metalsadman/pen/bGrzEXV?editors=1010

**Reply by kunle-e** [reply]
Yes, I see thank you. I was adding the position option in $q.dialog({}) that was referencing my custom component instead of the custom component's template.

**Reply by metalsadman** [reply]
make a prop in your custom dialog that sets it then pass it in Dialog plugin's `componentProps`.
```html
<!-- custom dialog template-->
<q-dialog :position="position">
```
```js
// custom dialog script
props: ['position']
```
```js
// parent call
q$.dialog({
  ...
  componentProps: {
    position: 'top'
  }
})
...
```
see update https://codepen.io/metalsadman/pen/bGrzEXV?editors=1010
